compiler/rustc_next_trait_solver/src/solve/assembly/structural_traits.rs

```diff
@@ -668,11 +668,12 @@ where
 {
     let cx = ecx.cx();
     let mut requirements = vec![];
-    requirements.extend(
+    requirements.extend(elaborate::elaborate(
```

Comment:
please add a comment why we elaborate here:
- it's not soundness critical
- however, we currently elaborate the super predicates for user-defined impls which destructures outlives requirements. So for the builtin and user-defined impl of `dyn Any: Any` to have the same region constraints (to enable us to merge them), we also elaborate to get all outlives constraints here

Reply:
Can do

Reply:
still relevant